addons/stock/models/stock_rule.py

```diff
@@ -541,8 +540,8 @@ def _search_rule(self, route_ids, packaging_id, product_id, warehouse_id, domain
         res = self.env['stock.rule']
         if route_ids:
             res = Rule.search(expression.AND([[('route_id', 'in', route_ids.ids)], domain]), order='route_sequence, sequence', limit=1)
-        if not res and packaging_id:
-            packaging_routes = packaging_id.route_ids
+        if not res and packaging_uom_id and packaging_uom_id.package_type_id.route_ids:
```

Comment:
```suggestion
        if not res and packaging_uom_id
```
Second half is already checked right afterwards :slightly_smiling_face:

Reply:
Nevermind that, forgot the obvious : `packaging_uom_id` has no `package_type_id` anyway, they're not packages.

Same remark applies to every call to a method that fetches rules to apply. I don't think that in any case we need to have the uom here, as it doesn't help on finding a specific route/rule.